onnx/defs/parser.h

```diff
@@ -72,22 +72,22 @@ class StringIntMap {
 class PrimitiveTypeNameMap : public StringIntMap<PrimitiveTypeNameMap> {
  public:
   PrimitiveTypeNameMap() : StringIntMap() {
-    map_["float"] = 1;
-    map_["uint8"] = 2;
-    map_["int8"] = 3;
-    map_["uint16"] = 4;
-    map_["int16"] = 5;
-    map_["int32"] = 6;
-    map_["int64"] = 7;
-    map_["string"] = 8;
-    map_["bool"] = 9;
-    map_["float16"] = 10;
-    map_["double"] = 11;
-    map_["uint32"] = 12;
-    map_["uint64"] = 13;
-    map_["complex64"] = 14;
-    map_["complex128"] = 15;
-    map_["bfloat16"] = 16;
+    map_["float"] = TensorProto_DataType_FLOAT;
+    map_["uint8"] = TensorProto_DataType_UINT8;
+    map_["int8"] = TensorProto_DataType_INT8;
+    map_["uint16"] = TensorProto_DataType_UINT16;
+    map_["int16"] = TensorProto_DataType_INT16;
+    map_["int32"] = TensorProto_DataType_INT32;
+    map_["int64"] = TensorProto_DataType_INT64;
+    map_["string"] = TensorProto_DataType_STRING;
+    map_["bool"] = TensorProto_DataType_BOOL;
+    map_["float16"] = TensorProto_DataType_FLOAT16;
+    map_["double"] = TensorProto_DataType_DOUBLE;
+    map_["uint32"] = TensorProto_DataType_UINT32;
+    map_["uint64"] = TensorProto_DataType_UINT64;
+    map_["complex64"] = TensorProto_DataType_COMPLEX64;
+    map_["complex128"] = TensorProto_DataType_COMPLEX128;
+    map_["bfloat16"] = TensorProto_DataType_BFLOAT16;
```

Comment:
Is it possible to reuse [TypeWrapper](https://github.com/onnx/onnx/blob/main/onnx/defs/data_type_utils.cc#L413) here to avoid duplication? cc @gramalingam

Reply:
Seems like `PrimitiveTypeNameMap` has the same feature as `DataTypeUtils` does. I think it would be better to remove `PrimitiveTypeNameMap` and `AttributeTypeNameMap`, merge `PrimitiveTypeNameMap` to `DataTypeUtils`, and create a similar class named `AttributeTypeUtils` that succeeds `AttributeTypeNameMap`.

Reply:
I'm working on other projects now, I will make a fix for this ASAP.

Reply:
Thanks for the fix. LGTM. Please note that code-freeze for the next release is in a few days. Please let us know whether we should merge this PR as-is (and do the extension discussed above as a separate PR) or wait for the extension. Thanks!